pkg/orchestration/wiring/aws/plugin.go

```diff
@@ -37,13 +37,14 @@ func Resource(resourceType voyager.ResourceType,
 	clusterServicePlanExternalID servicecatalog.PlanExternalID,
 	generateServiceEnvironment ServiceEnvironmentGenerator,
 	envResourcePrefix oap.EnvVarPrefix,
+	outputResourceContract bool,
```

Comment:
See my branch https://github.com/atlassian/voyager/compare/svccat-shapes

I think it is not do it or don't do it, it's which shape to output. Same here and in svc cat entangler. This is because some resources coming via OAP are not IAM accessible (not the case right now, but might be in the future).

Why don't you want to convert all of them at once? Why this flag per resource?

Reply:
This is WIP. Only UPS and DynamoDB are currently supported. Will add more resources later.

I decided to do incremental development and I stick to it. Sure you could do everything in one PR. Given that today I learned that this whole refactoring is much smaller change than I initially thought you might be right that this approach is overkill, but 🤷‍♂️ 

I'll see how many changes are required to support more resources. If not many I might support all remaining resources at once.

Reply:
Also just giving some reasoning on why we shouldn't use IsIamAccessible

Consider the case where we have a new set of providers that produce some sort of `KerberosyTokenThings`. Eventually our code will look like this:

```
	IsEnvVarBindable bool
	IsIAMAccessible  bool
	IsKerberosAccesssible bool
	IsTokenProducing  bool
	IsAnotherShape bool
	IsReallyReallyCool  bool
```

I think @kopper current implementation makes sense - a `SvcCatEntangler` should *always* produce a `BindableEnvVar`. Actually, when we refactor this more, a boolean will not be sufficient - we'll need to force the consumers of `SvcCatEntangler` to provide us the actual functions that can generate (or know about) the data:

```
GetEnvVarData  func (ServiceInstance) BindableEnvVarData
GetBinaryData  func (ServiceInstance) BindableBinaryData
GetSecretData  func (ServiceInstance) BindableSecretData
AdditionalShapes func (ServiceInstance) Shapes[]
```

Reply:
Note this code is only for (some) aws resources, so the number of bool flags will be limited to AWS-specific stuff.

On consumer side (EC2Compute, KubeCompute) I check if contract contains specific shape (in this case envvarbindable).

Reply:
I think all our OSB providers currently use it at the moment. A provider (UPS for example) doesn't seem to need to know about AWS to provide the flag